### PR TITLE
changes for docker xenial migration

### PR DESCRIPTION
**Pull request by challa**
changes include
 - New dependencies installation for lvm2
 - backward compatibility for trusty

### CODE REVIEW

File: jobs/docker/templates/bin/lvmvd_ctl.erb [hunk 1]
```diff
@@ -79,26 +79,82 @@ case $1 in
...
+              dpkg --unpack /var/vcap/packages/lvm2/liblvm2cmd2.02_2.02.133-1ubuntu10_amd64.deb
+	      set +e
+            fi
+
+            if ! dpkg -s dmeventd > /dev/null 2>&1 ; then
+              echo "Installing debian package dmeventd"
+	      set -e
+              dpkg --unpack /var/vcap/packages/lvm2/dmeventd_2_1.02.110-1ubuntu10_amd64.deb
+
+              # Now configure both liblvm2cmd2.02 and dmeventd which have cyclelic dependency
```
Comment (kamath-prasad) [inline jobs/docker/templates/bin/lvmvd_ctl.erb]: typo - `cyclic` ...

File: config/blobs.yml [hunk 2]
```diff
@@ -14,6 +14,26 @@ apt/lvm2/watershed_7_amd64.deb:
   size: 11356
   object_id: a72d0857-ae9b-4b2f-b5b6-725bee7006d9
   sha: 5c02014b7daecff6c0a6367996d2819d9b16745e
```
[the remark this answers is not in the document]
Comment (abh1kg) [reply config/blobs.yml]: They would if their older versions were approved via standard processes and we are adding newer ones here

File: jobs/docker/templates/bin/lvmvd_ctl.erb [hunk 3]
```diff
@@ -79,26 +79,82 @@ case $1 in
...
+              sudo dpkg --configure liblvm2cmd2.02 dmeventd
```
[the remark this answers is not in the document]
Comment (challa) [reply jobs/docker/templates/bin/lvmvd_ctl.erb]: yes sudo not required.

File: jobs/docker/templates/bin/lvmvd_ctl.erb [hunk 4]
```diff
@@ -79,26 +79,82 @@ case $1 in
...
+           echo "Installing Trusty packages for lvm2"
+
+           for package in libdevmapper-event1.02.1_2:1.02.77-6ubuntu2_amd64.deb watershed_7_amd64.deb lvm2_2.02.98-6ubuntu2_amd64.deb
+           do
+
+              # check if package already installed else install, incase package not found
+              # using dpkg -s for checking the installation which does not lock the /var/lib/dpkg/lock
+
+              if ! dpkg -s $package > /dev/null 2>&1 ; then
+                echo "Installing debian package $package"
```
[the remark this answers is not in the document]
Comment (challa) [reply jobs/docker/templates/bin/lvmvd_ctl.erb]: small change will  date too

File: jobs/docker/templates/bin/lvmvd_ctl.erb [hunk 5]
```diff
@@ -79,26 +79,82 @@ case $1 in
...
+
+    set +e
+    if [[ `lsb_release -cs` == "xenial" ]]; then
+
+            echo "Installing Xenial packages for lvm2"
+
+            if ! dpkg -s libdevmapper-event1.02.1 > /dev/null 2>&1 ; then
+              echo "Installing debian package libdevmapper-event1.02.1"
+	      set -e
+              dpkg -i /var/vcap/packages/lvm2/libdevmapper-event1.02.1_2_1.02.110-1ubuntu10_amd64.deb
```
Comment (subhankarc) [inline jobs/docker/templates/bin/lvmvd_ctl.erb]: Declare `/var/vcap/packages/lvm2` as a variable and use it

File: jobs/docker/templates/bin/lvmvd_ctl.erb (hunk 3 above)
Comment (subhankarc) [inline jobs/docker/templates/bin/lvmvd_ctl.erb]: Explicit `sudo` is not required I guess. Please remove if not required.

File: packages/lvm2/packaging [hunk 6]
```diff
@@ -8,6 +8,14 @@ set -u # report the usage of uninitialized variables
 # $BOSH_INSTALL_TARGET - where you copy/install files to be included in package
 
 # We see lock issues, maybe we can check these?
+
+# Packages for Xenial
```
Comment (subhankarc) [inline packages/lvm2/packaging]: Why do we keep all the packages in the VM? Why not check if it's xenial and copy selectively?

File: jobs/docker/templates/bin/lvmvd_ctl.erb (hunk 1 above)
Comment (challa) [reply jobs/docker/templates/bin/lvmvd_ctl.erb]: corrected typo

File: packages/lvm2/packaging (hunk 6 above)
Comment (challa) [reply packages/lvm2/packaging]: made changes to copy only necessary packages based on ubuntu version

Comment (subhankarc) [reply packages/lvm2/packaging]: thanks!

**Pull request merged**